### PR TITLE
decode: limits the number of decoded layers

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4125

Describe changes:
- fix bug by avoiding over recursion between `DecodeEthernet` and `DecodeMPLS`

Modifies #5904 by taking the comments into account (making the check earlier, adding comments)

suricata-verify-pr: 455

https://github.com/OISF/suricata-verify/pull/455

### CODE REVIEW

File: src/decode-events.c [hunk 1]
```diff
@@ -566,6 +566,10 @@ const struct DecodeEvents_ DEvents[] = {
             "decoder.nsh.unknown_payload",
             NSH_UNKNOWN_PAYLOAD,
     },
+    {
+            "decoder.too_many_layers",
```
Comment (victorjulien) [inline src/decode-events.c:570]: we also need a rule for this in `decoder-events.rules`

File: src/decode.h [hunk 2]
```diff
@@ -1161,6 +1166,18 @@ void DecodeUnregisterCounters(void);
 
 #define PKT_SET_SRC(p, src_val) ((p)->pkt_src = src_val)
 
+#define PKT_MAX_DECODED_LAYERS 16
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/decode.h:1169]: Something like `decoder.max-layers` or something (we already have the `decoder` object in our yaml)

File: src/decode-events.c (hunk 1 above)
Comment (catenacyber) [reply src/decode-events.c:570]: ok

File: src/decode.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/decode.h:1169]: ok

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/5921

**Pull request closed without merging**